service/src/main/java/be/bencouwberghs/timeless_songs/service/BandService.java

```
@@ -1,12 +1,14 @@
 package be.bencouwberghs.timeless_songs.service;
 
 import be.bencouwberghs.timeless_songs.model.Band;
+import be.bencouwberghs.timeless_songs.model.dto.BandDto;
 
 import java.util.List;
 
 public interface BandService {
     void addBand(Band band);
     void modifyBand(Band band);
+    void updateBand(Long id, BandDto bandDto);
```

Better not to do this.

Create a few separate classes, for example

Class `ValidateEntities` with at first method e.g. `validateBand(..)` in which you check if the needed properties are present. if not, let it throw an exception that you create e.g. UserInputException.
Call this from the controller and check for exception, if any then return this to the caller.

Another class MapperEntities with at first methods mapBandEntityToDto(..) and mapBandDtoToBandEntity(..)
Also use it in the controller.